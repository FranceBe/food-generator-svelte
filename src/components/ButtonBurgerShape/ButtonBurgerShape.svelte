<script>
    export let text
    export let onClick
    export let className = ''

    const onActionClick = e => {
        if (onClick) {
            onClick()
        } else {
            e.preventDefault()
        }
    }
</script>

<style>
    button {
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: flex-end;
        height: 50px;
        width: 50px;
        border-radius: 50px 50px 10px 10px;
        background: linear-gradient(0deg,
        #EBA789 15%,
        #50C92C 27%,
        #EE6663 47%,
        #F6FA19 45%,
        #50C92C 60%,
        #EBA789 79%);
        cursor: pointer;
    }
    button:focus {
        outline: none;
    }
    span {
        background-color: rgba(255, 255, 255, 0.7);
        padding: 5px;
        margin-bottom: 5px;
        font-size: 0.875rem;
    }
    .with-marge {
        margin-right: 10px;
    }
</style>

<button class={className} on:click={onActionClick}>
    <span>{text}</span>
</button>
