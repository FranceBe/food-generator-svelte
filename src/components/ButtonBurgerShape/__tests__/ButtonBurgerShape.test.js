import { render, fireEvent } from '@testing-library/svelte'
import ButtonBurgerShape from 'components/ButtonBurgerShape/ButtonBurgerShape.svelte'

describe('ButtonBurgerShape', () => {
  it('should match snapshot', () => {
    const { container } = render(ButtonBurgerShape, {
      props: {
        onClick: () => {
        },
        text: 'My button text',
      }
    })
    expect(container).toMatchSnapshot()
  })
  it('should render a button and a span', () => {
    const { container, getAllByText } = render(ButtonBurgerShape, {
      props: {
        onClick: () => {
        },
        text: 'My button text',
      }
    })
    expect(container.getElementsByTagName('button')).toHaveLength(1)
    expect(container.getElementsByTagName('span')).toHaveLength(1)
    expect(getAllByText('My button text').length).toBe(1)
  })
  it('should call onClick when button is clicked', async () => {
    const props = {
      onClick: jest.fn(),
      text: 'My button text',
    }
    const { getByText } = render(ButtonBurgerShape, {
      props,
    })
    const button = getByText('My button text')

    await fireEvent.click(button)
    expect(props.onClick).toHaveBeenCalledTimes(1)
    expect(props.onClick).toHaveBeenCalledWith()
  })
})
