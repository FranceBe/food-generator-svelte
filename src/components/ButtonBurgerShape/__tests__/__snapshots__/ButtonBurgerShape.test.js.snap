// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonBurgerShape should match snapshot 1`] = `
<body>
  <div>
    <button
      class=""
    >
      <span>
        My button text
      </span>
    </button>
  </div>
</body>
`;
